File: app/src/main/java/xyz/sleekstats/completist/view/GenreClickableSpan.java
package xyz.sleekstats.completist.view;

import android.graphics.Color;
import android.support.annotation.NonNull;
import android.text.TextPaint;
import android.text.style.ClickableSpan;
import android.view.View;

import xyz.sleekstats.completist.model.Genre;

public class GenreClickableSpan extends ClickableSpan {
    private final View.OnClickListener clickListener;
    private Genre content;

    public GenreClickableSpan(Genre content, View.OnClickListener clickListener) {
        super();
        this.content = content;
        this.clickListener= clickListener;
    }

    public void onClick(@NonNull View tv) {
        tv.setTag(content);
        clickListener.onClick(tv);
    }

    public void updateDrawState(@NonNull TextPaint ds) {
        ds.setColor(Color.parseColor("#689899"));
        ds.setUnderlineText(true); // set to false to remove underline
    }
}

File: app/src/main/java/xyz/sleekstats/completist/view/GenreClickableSpan.kt
package xyz.sleekstats.completist.view

import android.graphics.Color
import android.text.TextPaint
import android.text.style.ClickableSpan
import android.view.View
import xyz.sleekstats.completist.model.Genre

class GenreClickableSpan(private val content: Genre, private val clickListener: View.OnClickListener) : ClickableSpan() {

    override fun onClick(tv: View) {
        tv.tag = content
        clickListener.onClick(tv)
    }

    override fun updateDrawState(ds: TextPaint) {
        ds.color = Color.parseColor("#a68bf1")
    }

}